.gitlab-ci.yml

```diff
@@ -0,0 +1,203 @@
+        -dit
+        cpp_tango
+    - .travis/install_tango_idl.sh
+    - test ${STOCK_CPPZMQ} = "OFF" && .travis/install_cppzmq.sh || true
+  script:
+    - set -e
+    - .travis/run.sh
+    - .travis/install_tango.sh
+    - .travis/install_tango_admin.sh
+    - .travis/test.sh
+    - set +e
+    - test ${SONAR_SCANNER} = "ON" && .travis/sonar.sh || true
+  after_script:
+    - docker stop cpp_tango
+    - mkdir -p build/cpp_test_suite/test_results
+    - gzip -r build/cpp_test_suite/test_results
+  artifacts:
+    when: always
+    expire_in: 1h
+    paths:
+      - build/cpp_test_suite/test_results
+      - build/cpp_test_suite/*/core.*
+
+abi-api-compliance-check:
+  variables:
+    <<: *variables
+    RUN_TESTS: "OFF"
+  only:
+    - merge_requests
+  image: debian:10
+  before_script:
+    - apt-get update && apt-get install -y git docker.io
+    - .travis/before_install.sh
+    - docker build -t cpp_tango .travis/gcc-latest
+    - >
+        docker run
+        --rm
+        --name cpp_tango
+        --user root
+        -v `pwd`:/home/tango/src
+        -v `pwd`/idl:/home/tango/idl
+        -dit
+        cpp_tango
+    - .travis/install_tango_idl.sh
+  script:
+    - >
+        docker exec
+        -w /home/tango/src
+        -e CI_TARGET_BRANCH=${CI_MERGE_REQUEST_TARGET_BRANCH_NAME}
+        -e CMAKE_BUILD_PARALLEL_LEVEL=$(nproc)
+        cpp_tango
+        .travis/check-ABI-API-compliance.sh || true
+  after_script:
+    - docker stop cpp_tango
+    - gzip -r compat_reports
+  artifacts:
+    when: always
+    expire_in: 1h
+    paths:
+      - compat_reports
+
+llvm-latest:
+  <<: *job-template
+  variables:
+    <<: *variables
+    OS_TYPE: llvm-latest
+    RUN_TESTS: "OFF"
+    WARNINGS_AS_ERRORS: "ON"
+    TANGO_USE_USING_NAMESPACE: "OFF"
+    USE_PCH: "OFF"
+
+gcc-latest:
+  <<: *job-template
+  variables:
+    <<: *variables
+    OS_TYPE: gcc-latest
+    RUN_TESTS: "OFF"
+    WARNINGS_AS_ERRORS: "ON"
+    TANGO_USE_USING_NAMESPACE: "OFF"
+    USE_PCH: "OFF"
+
+ubuntu-20.04:
+  <<: *job-template
+  variables:
+    <<: *variables
+    OS_TYPE: ubuntu-20.04
+
+debian10:
+  <<: *job-template
+  variables:
+    <<: *variables
+    TANGO_ENABLE_COVERAGE: "ON"
+    OS_TYPE: debian10
+  after_script:
+    - mkdir -p build/cpp_test_suite/test_results
+    - gzip -r build/cpp_test_suite/test_results
+    - mkdir coverage
+    - >
+        docker exec --workdir /home/tango/src cpp_tango
+        gcovr --filter '^cppapi/' --filter '^log4tango/' -j$(nproc)
+        --xml --output coverage.xml
+    - >
+        docker exec --workdir /home/tango/src cpp_tango
+        gcovr --filter '^cppapi/' --filter '^log4tango/' -j$(nproc)
+        --html-details --output coverage/coverage.html
+    - >
+        docker exec --workdir /home/tango/src cpp_tango
+        gcovr --filter '^cppapi/' --filter '^log4tango/' -j$(nproc)
+    - tar czf coverage.tar.gz coverage
+    - docker stop cpp_tango
+  artifacts:
+    when: always
+    expire_in: 1h
```

Comment:
isn't 1h a bit too short?

Reply:
Right. What do you propose? Maybe 1 day would be ok? I'd rather not abuse GL-provided storage with old test logs and core dumps. Also note that artifacts from the last run of each job are kept indefinitely, regardless of this setting.

Reply:
I would try not to abuse with core files, which can be big indeed.
But I think having several days would be great (so we can launch something on Friday evening and come back on Monday or Tuesday and get the results).
I'm not sure to understand fully your sentence "Also note that artifacts from the last run of each job are kept indefinitely, regardless of this setting."
What happens if someone makes a new commit in another branch? Will this one be considered as the last run?
Or is it the artifacts from the last run from each branch which are kept?

Reply:
Why not just use the default for now? We need to buy more project space anyway soon.

Reply:
I did as Thomas suggested and removed explicit setting of expiration time.

> What happens if someone makes a new commit in another branch? Will this one be considered as the last run?
Or is it the artifacts from the last run from each branch which are kept?

Per my understanding, it keeps all artifacts from all jobs in the last pipeline of each branch, MR and tag:
https://docs.gitlab.com/ee/ci/yaml/#artifactsexpire_in
https://gitlab.com/gitlab-org/gitlab/-/issues/16267
https://gitlab.com/gitlab-org/gitlab/-/issues/229936